File: src/bot/commands/skip.ts
import type { Message } from 'discord.js-selfbot-v13';
import { getVideoStreamer } from '../../stream/video-streamer.js';
import { getNextEpisode, formatDuration } from '../../plex/library.js';
import plexClient from '../../plex/client.js';

export async function skipCommand(message: Message, _args: string[]): Promise<void> {
  if (!message.guild) {
    await message.edit('❌ This command can only be used in a server');
    return;
  }

  const guildId = message.guild.id;
  const videoStreamer = getVideoStreamer();
  const session = videoStreamer.getSession(guildId);

  if (!session) {
    await message.edit('❌ Nothing is currently playing');
    return;
  }

  if (session.mediaItem.type !== 'episode') {
    await message.edit('❌ Skip is only available for TV shows');
    return;
  }

  await message.edit('⏭️ Loading next episode...');

  const nextEpisode = await getNextEpisode(session.mediaItem);

  if (!nextEpisode) {
    await message.edit('❌ No next episode available (end of series or season)');
    return;
  }

  try {
    const streamInfo = await plexClient.getDirectStreamUrl(nextEpisode.ratingKey);
    if (!streamInfo) {
      throw new Error('Could not get stream URL for next episode');
    }

    await videoStreamer.startStream(
      guildId,
      session.channelId,
      nextEpisode,
      streamInfo.url,
      0
    );

    const season = nextEpisode.parentIndex ? `S${String(nextEpisode.parentIndex).padStart(2, '0')}` : '';
    const episode = nextEpisode.index ? `E${String(nextEpisode.index).padStart(2, '0')}` : '';
    const title = `${nextEpisode.grandparentTitle || ''} ${season}${episode} - ${nextEpisode.title}`;
    const duration = nextEpisode.duration ? formatDuration(nextEpisode.duration) : 'Unknown';

    await message.edit(`⏭️ **Now Streaming:** ${title}\n⏱️ Duration: ${duration}`);
  } catch (error) {
    console.error('[Skip] Error:', error);
    await message.edit(`❌ Failed to skip: ${error instanceof Error ? error.message : 'Unknown error'}`);
  }
}

File: selfbot/commands/skip.py
"""Skip command — skip to next episode via IPC."""

from __future__ import annotations

from typing import TYPE_CHECKING
import discord

from ..ipc import ipc

if TYPE_CHECKING:
    from ..client import SchroStreamBot


async def skip_command(bot: SchroStreamBot, message: discord.Message, args: list[str]) -> None:
    if not message.guild:
        await message.channel.send("❌ This command can only be used in a server")
        return

    status_msg = await message.channel.send("⏭️ Loading next episode...")

    result = await ipc.plex_skip(str(message.guild.id))

    if result.get("success"):
        title = result.get("title", "Next episode")
        duration = result.get("duration")
        dur_str = f"\n⏱️ Duration: {duration}" if duration else ""
        await status_msg.edit(content=f"⏭️ **Now Streaming:** {title}{dur_str}")
    else:
        await status_msg.edit(content=f"❌ {result.get('error', 'No next episode available')}")
